src/render/renderer.cpp

```diff
@@ -0,0 +1,132 @@
+#include "render/renderer.h"
+
+#include <iostream>
+#include <limits>
+
+namespace render
+{
+
+Renderer::Renderer( const float fov,
+                    const int imageWidth,
```

Comment:
maybe it will be better to change imageWidth and imageHeight to unsigned int or size_t in all the code